packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -663,11 +703,35 @@ export default class GoogleCalendarService implements Calendar {
   ): Promise<calendar_v3.Schema$FreeBusyResponse> {
     if (shouldServeCache === false) return await this.fetchAvailability(args);
     const calendarCache = await CalendarCache.init(null);
-    const cached = await calendarCache.getCachedAvailability(this.credential.id, args);
-    if (cached) return cached.value as unknown as calendar_v3.Schema$FreeBusyResponse;
+
+    const cached = await calendarCache.getCachedAvailability({
+      credentialId: this.credential.id,
+      delegationCredentialId: this.credential.delegatedToId ?? null,
+      userId: this.credential.userId,
+      args,
+    });
+
+    if (cached) {
+      log.debug("[Cache Hit] Returning cached freebusy result", safeStringify({ cached, args }));
+      return cached.value as unknown as calendar_v3.Schema$FreeBusyResponse;
+    }
+    log.debug("[Cache Miss] Fetching freebusy result", safeStringify({ args }));
     return await this.fetchAvailability(args);
   }
 
+  getValidCalendars<T extends { id?: string | null }>(cals: T[]) {
+    return cals.filter((cal): cal is T & { id: string } => !!cal.id);
+  }
+
+  filterPrimaryCalendar(cals: calendar_v3.Schema$CalendarListEntry[]) {
```

Comment:
Gives primary calendar, that is either explicitly marked as primary by Google Calendar or use the first one.